src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -146,8 +146,13 @@ trait Namers extends MethodSynthesis {
     }
 
     def setPrivateWithin[T <: Symbol](tree: Tree, sym: T, mods: Modifiers): T =
-      if (sym.isPrivateLocal || !mods.hasAccessBoundary) sym
-      else sym setPrivateWithin typer.qualifyingClass(tree, mods.privateWithin, packageOK = true, immediate = false)
+      if (sym.isPrivateLocal) sym
+      else {
+        val qualClass = if (mods.hasAccessBoundary)
+          typer.qualifyingClass(tree, mods.privateWithin, packageOK = true, immediate = false)
+        else NoSymbol
+          sym setPrivateWithin qualClass
```

Comment:
Indentation is off here. What happens to synthetic apply of case class? I'd expect it to have same access as the constructor.

Reply:
I remembered "subvert": https://github.com/scala/bug/issues/7884